amaranth/vendor/xilinx.py

```diff
@@ -343,14 +343,20 @@ def vendor_toolchain(self):
 
     _symbiflow_part_map = {
         "xc7a35ticsg324-1L": "xc7a35tcsg324-1", # Arty-A7
```

Comment:
Does this map just cut off the last letter of the speed grade? (cc @mwkmwkmwk, I don't quite have Xilinx part numbering scheme memorized)

Reply:
yes, specifically cuts the "low-power" designation

Reply:
Ok yeah we can definitely use a function here.

Reply:
I don't know exactly how symbiflow part naming is done, but having a map here is probably safest.

Reply:
The symbiflow [part db](https://github.com/chipsalliance/f4pga/blob/main/f4pga/flows/part_db.yml) seems to just drop the speed grade from the Vivado part name.

Reply:
I turned this into a re.sub call.